tools/autograd/gen_variable_type.py

```diff
@@ -599,7 +599,41 @@ def emit_buffers(buffers):
         return res
 
     def emit_record_trace(env, declaration):
+
+        # Note [clang-802.0.42 tuple overload bug]
```

Comment:
I don't see this note as being very useful for any reading the code. It's all about the previous implementation, which you don't see. The new implementation is simple and makes sense. There are a very large number of alternative implementations that may be reasonable, and we don't explain the decision for not using each of them.

This seems more appropriate in the commit message.

Reply:
Well, it's in the commit message, so we can always delete this note later :)